### PR TITLE
chore: enforce formatting and linting

### PR DESCRIPTION
**Pull request by erezrokah**
This adds linting and formatting and enforces those in CI

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,19 +1,26 @@
 /* Generates a sitemap */
+const { env } = require('process')
+
 const makeSitemap = require('./make_sitemap')
 
+const getBuildDir = ({ inputs, constants }) => {
```
Comment (erezrokah) [inline index.js:6]: As a result of code complexity rule

File: make_sitemap.js [hunk 2]
```diff
@@ -1,40 +1,57 @@
...
+const util = require('util')
+
+const globby = require('globby')
 const mkdirp = require('mkdirp')
 const sm = require('sitemap')
-const globby = require('globby')
 
-module.exports = async function makeSitemap(opts = {}) {
-  const { distPath, fileName, homepage, exclude, prettyURLs, trailingSlash, failBuild } = opts
+const getPaths = async ({ distPath, exclude }) => {
```
Comment (erezrokah) [inline make_sitemap.js:9]: As a result of code complexity rule

File: make_sitemap.js [hunk 3]
```diff
@@ -1,40 +1,57 @@
...
-      urlPath = urlPath.replace(/\/index\.html$/, '').replace(/\.html$/, '')
-
-      if (trailingSlash) {
-        urlPath += urlPath.endsWith("/") ? "" : "/";
-      }
-    }
+  return paths
+}
+
+const getUrlFromFile = ({ file, distPath, prettyURLs, trailingSlash }) => {
```
Comment (erezrokah) [inline make_sitemap.js:18]: As a result of code complexity rule

File: tests/sitemap.test.js [hunk 4]
```diff
@@ -1,32 +1,31 @@
...
 const SITEMAP_OUTPUT = path.resolve(BUILDPATH, 'sitemap.xml')
 
-test.afterEach.always(t => {
+test.afterEach.always(() => {
   // Cleanup runs after each test
   rimraf.sync(SITEMAP_OUTPUT)
 })
 
 test.serial('Creates Sitemap with all html files', async (t) => {
-  let xmlData
```
Comment (erezrokah) [inline tests/sitemap.test.js:17]: I removed the `try/catch` from all tests - a thrown error fails the test anyway

File: index.js (hunk 1 above)
Comment (ehmicky) [reply index.js:6]: Yes, that's actually probably a good thing to split this function.

**Review by ehmicky — APPROVED** [review]
The code looks so much better :tada:

**Pull request merged**